### DISCUSSION TITLE
webauthn4j graalvm native compilation issues

### DISCUSSION THREAD

**Discussion by madocx** [webauthn4j/webauthn4j#853] [Q&A]
Working on a webauthn poc, utilizing webauthn4j (https://github.com/webauthn4j/webauthn4j) in a quarkus application. Everything works perfectly running in a JVM, but after compiling natively using graalvm, the library fails verification of the signature. Specifically, com.webauthn4j.validator.AssertionSignatureValidator.verifySignature().

Under the covers, it seems webauthn4j is using JCASecurityServices (please correct me if I'm wrong). However, none of the recommendations here seem to help : https://www.graalvm.org/latest/reference-manual/native-image/dynamic-features/JCASecurityServices/

Has anyone else faced something like this, or have any recommendations on what might possibly be causing it? This is the first time I've run into something like this where it natively compiles without any issues, but just behaves entirely differently (and incorrectly) at runtime.

**Comment by ynojima** [comment]
If the signature doesn't matches, I assume it is deserialization issue.
webauthn4j use jackson, which havily uses reflection. Did you exclude webauthn4j classes not to be deleted by graalvm?
